### PR TITLE
Removed unnecessary assignments

### PR DESCRIPTION
**Pull request by Rubonnek**
`cppcheck` found variables whose previous values were never used before an assignment.

This pull request mitigates the following `cppcheck` messages:
```
[core/bind/core_bind.cpp:1409] -> [core/bind/core_bind.cpp:1413]: (style, inconclusive) Variable 'err' is reassigned a value before the old one has been used if variable is no semaphore variable.
[core/io/file_access_compressed.cpp:371] -> [core/io/file_access_compressed.cpp:377]: (style, inconclusive) Variable 'block_size' is reassigned a value before the old one has been used if variable is no semaphore variable.
[drivers/gles2/rasterizer_gles2.cpp:657] -> [drivers/gles2/rasterizer_gles2.cpp:658]: (style, inconclusive) Variable 'r_gl_internal_format' is reassigned a value before the old one has been used if variable is no semaphore variable.
[editor/code_editor.cpp:261] -> [editor/code_editor.cpp:262]: (style, inconclusive) Variable 'r_col' is reassigned a value before the old one has been used if variable is no semaphore variable.
[editor/editor_asset_installer.cpp:103] -> [editor/editor_asset_installer.cpp:108]: (style) Variable 'ret' is reassigned a value before the old one has been used.
[editor/export_template_manager.cpp:198] -> [editor/export_template_manager.cpp:199]: (style) Variable 'ret' is reassigned a value before the old one has been used.
[editor/export_template_manager.cpp:260] -> [editor/export_template_manager.cpp:268]: (style) Variable 'ret' is reassigned a value before the old one has been used.
[editor/export_template_manager.cpp:268] -> [editor/export_template_manager.cpp:269]: (style) Variable 'ret' is reassigned a value before the old one has been used.
[editor/export_template_manager.cpp:269] -> [editor/export_template_manager.cpp:290]: (style) Variable 'ret' is reassigned a value before the old one has been used.
[editor/filesystem_dock.cpp:1468] -> [editor/filesystem_dock.cpp:1471]: (style) Variable 'efsd' is reassigned a value before the old one has been used.
[editor/import/editor_import_collada.cpp:1587] -> [editor/import/editor_import_collada.cpp:1594]: (style) Variable 'joint_src' is reassigned a value before the old one has been used.
[editor/import/editor_import_collada.cpp:2319] -> [editor/import/editor_import_collada.cpp:2320]: (style, inconclusive) Variable 'anim' is reassigned a value before the old one has been used if variable is no semaphore variable.
[editor/plugins/particles_editor_plugin.cpp:273] -> [editor/plugins/particles_editor_plugin.cpp:274]: (style, inconclusive) Variable 'ofs' is reassigned a value before the old one has been used if variable is no semaphore variable.
[editor/script_create_dialog.cpp:534] -> [editor/script_create_dialog.cpp:547]: (style) Variable 'gc' is reassigned a value before the old one has been used.
[editor/script_create_dialog.cpp:535] -> [editor/script_create_dialog.cpp:552]: (style) Variable 'vb' is reassigned a value before the old one has been used.
[editor/script_create_dialog.cpp:536] -> [editor/script_create_dialog.cpp:554]: (style) Variable 'hb' is reassigned a value before the old one has been used.
[editor/script_create_dialog.cpp:537] -> [editor/script_create_dialog.cpp:555]: (style) Variable 'l' is reassigned a value before the old one has been used.
[editor/script_create_dialog.cpp:541] -> [editor/script_create_dialog.cpp:574]: (style) Variable 'pc' is reassigned a value before the old one has been used.
[editor/script_create_dialog.cpp:539] -> [editor/script_create_dialog.cpp:581]: (style) Variable 'empty_h' is reassigned a value before the old one has been used.
[editor/script_create_dialog.cpp:540] -> [editor/script_create_dialog.cpp:586]: (style) Variable 'empty_v' is reassigned a value before the old one has been used.
[main/main.cpp:893] -> [main/main.cpp:895]: (style) Variable 'show_logo' is reassigned a value before the old one has been used.
[main/tests/test_string.cpp:805] -> [main/tests/test_string.cpp:820]: (style) Variable 'success' is reassigned a value before the old one has been used.
[modules/gdnative/gdnative.cpp:250] -> [modules/gdnative/gdnative.cpp:252]: (style, inconclusive) Variable 'error' is reassigned a value before the old one has been used if variable is no semaphore variable.
[modules/gdscript/gd_compiler.cpp:1454] -> [modules/gdscript/gd_compiler.cpp:1462]: (style) Variable 'gdfunc' is reassigned a value before the old one has been used.
[modules/gdscript/gd_parser.cpp:2384] -> [modules/gdscript/gd_parser.cpp:2386]: (style) Variable 'subexpr' is reassigned a value before the old one has been used.
[modules/gdscript/gd_parser.cpp:3138] -> [modules/gdscript/gd_parser.cpp:3140]: (style) Variable 'defval' is reassigned a value before the old one has been used.
[modules/gdscript/gd_parser.cpp:3878] -> [modules/gdscript/gd_parser.cpp:3880]: (style) Variable 'subexpr' is reassigned a value before the old one has been used.
[modules/gdscript/gd_parser.cpp:4038] -> [modules/gdscript/gd_parser.cpp:4040]: (style) Variable 'subexpr' is reassigned a value before the old one has been used.
[modules/gdscript/gd_parser.cpp:4106] -> [modules/gdscript/gd_parser.cpp:4108]: (style) Variable 'subexpr' is reassigned a value before the old one has been used.
[modules/gdscript/gd_tokenizer.cpp:1190] -> [modules/gdscript/gd_tokenizer.cpp:1192]: (style) Variable 'b' is reassigned a value before the old one has been used.
[modules/nativescript/api_generator.cpp:158] -> [modules/nativescript/api_generator.cpp:161]: (style) Variable 'is_reference' is reassigned a value before the old one has been used.
[modules/openssl/stream_peer_openssl.cpp:44] -> [modules/openssl/stream_peer_openssl.cpp:50]: (style) Variable 'common_name_loc' is reassigned a value before the old one has been used.
[modules/openssl/stream_peer_openssl.cpp:45] -> [modules/openssl/stream_peer_openssl.cpp:55]: (style) Variable 'common_name_entry' is reassigned a value before the old one has been used.
[modules/openssl/stream_peer_openssl.cpp:46] -> [modules/openssl/stream_peer_openssl.cpp:60]: (style) Variable 'common_name_asn1' is reassigned a value before the old one has been used.
[modules/openssl/stream_peer_openssl.cpp:47] -> [modules/openssl/stream_peer_openssl.cpp:64]: (style) Variable 'common_name_str' is reassigned a value before the old one has been used.
[modules/pbm/bitmap_loader_pbm.cpp:98] -> [modules/pbm/bitmap_loader_pbm.cpp:100]: (style, inconclusive) Variable 'w' is reassigned a value before the old one has been used if variable is no semaphore variable.
[modules/theora/video_stream_theora.cpp:454] -> [modules/theora/video_stream_theora.cpp:455]: (style, inconclusive) Variable 'pp_level' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/android/audio_driver_jandroid.cpp:82] -> [platform/android/audio_driver_jandroid.cpp:83]: (style) Variable 'latency' is reassigned a value before the old one has been used.
[platform/android/export/export.cpp:3027] -> [platform/android/export/export.cpp:3037]: (style, inconclusive) Variable 'err' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/android/export/export.cpp:3037] -> [platform/android/export/export.cpp:3049]: (style, inconclusive) Variable 'err' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/iphone/audio_driver_iphone.cpp:52] -> [platform/iphone/audio_driver_iphone.cpp:60]: (style, inconclusive) Variable 'comp' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/iphone/audio_driver_iphone.cpp:49] -> [platform/iphone/audio_driver_iphone.cpp:63]: (style, inconclusive) Variable 'result' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/osx/audio_driver_osx.cpp:56] -> [platform/osx/audio_driver_osx.cpp:66]: (style, inconclusive) Variable 'comp' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/osx/joypad_osx.cpp:220] -> [platform/osx/joypad_osx.cpp:223]: (style, inconclusive) Variable 'refCF' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/osx/joypad_osx.cpp:292] -> [platform/osx/joypad_osx.cpp:298]: (style, inconclusive) Variable 'refCF' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/osx/joypad_osx.cpp:337] -> [platform/osx/joypad_osx.cpp:338]: (style, inconclusive) Variable 'array' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/windows/os_windows.cpp:720] -> [platform/windows/os_windows.cpp:721]: (style, inconclusive) Variable 'hDropInfo' is reassigned a value before the old one has been used if variable is no semaphore variable.
[platform/x11/context_gl_x11.cpp:88] -> [platform/x11/context_gl_x11.cpp:92]: (style) Variable 'glXCreateContextAttribsARB' is reassigned a value before the old one has been used.
[platform/x11/os_x11.cpp:194] -> [platform/x11/os_x11.cpp:195]: (style) Variable 'imvalret' is reassigned a value before the old one has been used.
[platform/x11/os_x11.cpp:2154] -> [platform/x11/os_x11.cpp:2155]: (style) Variable 'classHint' is reassigned a value before the old one has been used.
[scene/2d/polygon_2d.cpp:146] -> [scene/2d/polygon_2d.cpp:148]: (style, inconclusive) Variable 'tex_size' is reassigned a value before the old one has been used if variable is no semaphore variable.
[scene/3d/vehicle_body.cpp:562] -> [scene/3d/vehicle_body.cpp:568]: (style, inconclusive) Variable 'rel_vel' is reassigned a value before the old one has been used if variable is no semaphore variable.
[scene/gui/text_edit.cpp:3824] -> [scene/gui/text_edit.cpp:3827]: (style) Variable 'line' is reassigned a value before the old one has been used.
[scene/gui/tree.cpp:1460] -> [scene/gui/tree.cpp:1462]: (style, inconclusive) Variable 'selected_item' is reassigned a value before the old one has been used if variable is no semaphore variable.
[scene/gui/video_player.cpp:398] -> [scene/gui/video_player.cpp:402]: (style, inconclusive) Variable 'loops' is reassigned a value before the old one has been used if variable is no semaphore variable.
[scene/resources/primitive_meshes.cpp:1021] -> [scene/resources/primitive_meshes.cpp:1025]: (style) Variable 'offset_back' is reassigned a value before the old one has been used.
[servers/physics/broad_phase_octree.cpp:125] -> [servers/physics/broad_phase_octree.cpp:127]: (style, inconclusive) Variable 'pair_callback' is reassigned a value before the old one has been used if variable is no semaphore variable.
[servers/physics/space_sw.cpp:477] -> [servers/physics/space_sw.cpp:478]: (style) Variable 'obj' is reassigned a value before the old one has been used.
[servers/physics/space_sw.cpp:603] -> [servers/physics/space_sw.cpp:604]: (style) Variable 'cbkptr' is reassigned a value before the old one has been used.
[servers/physics/space_sw.cpp:601] -> [servers/physics/space_sw.cpp:605]: (style) Variable 'cbkres' is reassigned a value before the old one has been used.
[servers/physics_2d/joints_2d_sw.cpp:411] -> [servers/physics_2d/joints_2d_sw.cpp:412]: (style, inconclusive) Variable '_b' is reassigned a value before the old one has been used if variable is no semaphore variable.
[servers/physics_2d/space_2d_sw.cpp:537] -> [servers/physics_2d/space_2d_sw.cpp:538]: (style) Variable 'cbkptr' is reassigned a value before the old one has been used.
[servers/physics_2d/space_2d_sw.cpp:535] -> [servers/physics_2d/space_2d_sw.cpp:539]: (style) Variable 'cbkres' is reassigned a value before the old one has been used.
```

### CODE REVIEW

File: servers/physics_2d/joints_2d_sw.cpp [hunk 1]
```diff
@@ -408,8 +408,7 @@ bool GrooveJoint2DSW::setup(real_t p_step) {
 	//joint->bias = cpvclamp(cpvmult(delta, -joint->constraint.biasCoef*dt_inv), joint->constraint.maxBias);
 
 	Vector2 delta = (B->get_transform().get_origin() + rB) - (A->get_transform().get_origin() + rA);
-	real_t _b = get_bias();
-	_b = 0.001;
+	real_t _b = 0.001;
```
Comment (reduz) [inline servers/physics_2d/joints_2d_sw.cpp]: this kinda looks like a bug..

File: editor/export_template_manager.cpp [hunk 2]
```diff
@@ -195,7 +195,7 @@ void ExportTemplateManager::_install_from_file(const String &p_file) {
 			data.resize(info.uncompressed_size);
 
 			//read
-			ret = unzOpenCurrentFile(pkg);
+			unzOpenCurrentFile(pkg);
```
Comment (akien-mga) [inline editor/export_template_manager.cpp]: Note that this will trigger another warning about return argument unused. It looks like code analysis tool want to force us to do result-checking even if we don't care about it :)

File: modules/openssl/stream_peer_openssl.cpp [hunk 3]
```diff
@@ -41,27 +41,22 @@ bool StreamPeerOpenSSL::_match_host_name(const char *name, const char *hostname)
 
 Error StreamPeerOpenSSL::_match_common_name(const char *hostname, const X509 *server_cert) {
 
-	int common_name_loc = -1;
-	X509_NAME_ENTRY *common_name_entry = NULL;
-	ASN1_STRING *common_name_asn1 = NULL;
-	char *common_name_str = NULL;
```
Comment (akien-mga) [inline modules/openssl/stream_peer_openssl.cpp]: Poor C-style developers... :(

File: modules/theora/video_stream_theora.cpp [hunk 4]
```diff
@@ -451,7 +451,6 @@ void VideoStreamPlaybackTheora::set_file(const String &p_file) {
 					ti.frame_width, ti.frame_height, ti.pic_x, ti.pic_y);
 		th_decode_ctl(td, TH_DECCTL_GET_PPLEVEL_MAX, &pp_level_max,
 				sizeof(pp_level_max));
-		pp_level = pp_level_max;
 		pp_level = 0;
```
Comment (akien-mga) [inline modules/theora/video_stream_theora.cpp]: For the reference, this was added by @reduz in 7ba484bc74. The above removed line was from 0b806ee0fc, the birth of this repo, so we can't go further.

File: scene/3d/vehicle_body.cpp [hunk 5]
```diff
@@ -559,13 +559,13 @@ void VehicleBody::_resolve_single_bilateral(PhysicsDirectBodyState *s, const Vec
...
 
-	real_t rel_vel = jac.getRelativeVelocity(
+	jac.getRelativeVelocity(
 			s->get_linear_velocity(),
 			s->get_transform().basis.transposed().xform(s->get_angular_velocity()),
 			b2lv,
 			b2trans.xform(b2av));
 
-	rel_vel = normal.dot(vel);
+	real_t rel_vel = normal.dot(vel);
```
Comment (akien-mga) [inline scene/3d/vehicle_body.cpp]: I think you're hiding a bug here. `getRelativeVelocity()` is just a getter so not assigning its result to a variable is pointless.

I guess what was intended is `rel_vel = normal.dot(rel_vel);`.

File: modules/pbm/bitmap_loader_pbm.cpp [hunk 6]
```diff
@@ -95,7 +95,7 @@ static bool _get_token(FileAccessRef &f, uint8_t &saved, PoolVector<uint8_t> &r_
 				resized = true;
 			}
 			if (resized) {
-				w = PoolVector<uint8_t>::Write();
+				PoolVector<uint8_t>::Write();
 				r_token.resize(token_max);
 				w = r_token.write();
```
Comment (karroffel) [inline modules/pbm/bitmap_loader_pbm.cpp]: This is a bug. `PoolVector<uint8_t>::Write()` has an [`operator=` method](https://github.com/godotengine/godot/blob/master/core/dvector.h#L329).

So here this actually performs side effects and is not just a weird mistake.

File: editor/export_template_manager.cpp (hunk 2 above)
Comment (Rubonnek) [inline editor/export_template_manager.cpp]: That's right. There's a reason why it's configurable. I mostly pay attention to this kind of `cppcheck` message only because it sometimes help bring some possible bugs to the surface like the one reduz pointed out. I don't mind undoing any of these changes at all if they are unwanted.

File: modules/openssl/stream_peer_openssl.cpp (hunk 3 above)
Comment (Rubonnek) [inline modules/openssl/stream_peer_openssl.cpp]: :smile: Should I undo this change?

**Review by karroffel — DISMISSED** [review]

File: servers/physics_2d/joints_2d_sw.cpp (hunk 1 above)
Comment (akien-mga) [inline servers/physics_2d/joints_2d_sw.cpp]: This appears to be like this since Godot was open sourced (first commit).

I suggest you change that code to:
```
// FIXME: We used to do this assignment and then override it with 0.001 right after,
// investigate why.
//real_t _b = get_bias();
real_t _b = 0.001;
```

File: modules/pbm/bitmap_loader_pbm.cpp (hunk 6 above)
Comment (Rubonnek) [inline modules/pbm/bitmap_loader_pbm.cpp]: Got it. I'll undo this change.

File: servers/physics_2d/joints_2d_sw.cpp (hunk 1 above)
Comment (Rubonnek) [inline servers/physics_2d/joints_2d_sw.cpp]: Will do.

File: modules/openssl/stream_peer_openssl.cpp (hunk 3 above)
Comment (akien-mga) [inline modules/openssl/stream_peer_openssl.cpp]: Hehe no it's good :)

File: editor/export_template_manager.cpp (hunk 2 above)
Comment (akien-mga) [inline editor/export_template_manager.cpp]: I'm not sure, I guess it's good how you changed it. If we don't care about the return value, let's not assign it. If we're bothered by that other warning, we can disable it or start caring and check the return value (I guess we should do the latter :)).

File: modules/pbm/bitmap_loader_pbm.cpp (hunk 6 above)
Comment (akien-mga) [inline modules/pbm/bitmap_loader_pbm.cpp]: Please add a comment to mention why it's done this way then, to ensure we don't try to remove it again.

File: platform/android/audio_driver_jandroid.cpp [hunk 7]
```diff
@@ -79,8 +79,7 @@ Error AudioDriverAndroid::init() {
 	JNIEnv *env = ThreadAndroid::get_env();
 	int mix_rate = GLOBAL_DEF("audio/mix_rate", 44100);
 
-	int latency = GLOBAL_DEF("audio/output_latency", 25);
-	latency = 50;
+	int latency = GLOBAL_DEF("audio/output_latency", 50);
```
Comment (akien-mga) [inline platform/android/audio_driver_jandroid.cpp]: Looks like you're fixing a bug here... or introducing a new one, we'll see :P

File: scene/3d/vehicle_body.cpp (hunk 5 above)
Comment (akien-mga) [inline scene/3d/vehicle_body.cpp]: I think for now it's best if you leave the code as it was, and add a FIXME comment. My guess *might* be correct, but it's best fixed in its own commit for clarity anyway, after checking that it doesn't not break VehicleBody.

File: editor/export_template_manager.cpp (hunk 2 above)
Comment (Rubonnek) [inline editor/export_template_manager.cpp]: :D

File: scene/3d/vehicle_body.cpp (hunk 5 above)
Comment (Rubonnek) [inline scene/3d/vehicle_body.cpp]: I'll add the comment then. Thanks.

Comment (Zireael07) [inline scene/3d/vehicle_body.cpp]: @akien-mga: VehicleBody is already broken in 3.0 :/

Comment (akien-mga) [inline scene/3d/vehicle_body.cpp]: Well that can be tested on 2.1 too, that code did not change since 2014.

File: platform/android/audio_driver_jandroid.cpp (hunk 7 above)
Comment (Rubonnek) [inline platform/android/audio_driver_jandroid.cpp]: Haha, I guess we will. :P

Comment (Rubonnek) [inline platform/android/audio_driver_jandroid.cpp]: It seems like I'm introducing a bug here instead.  I am seeing `GLOBAL_DEF("audio/output_latency", 25)` way too many times everywhere else:

```
servers/audio/audio_driver_dummy.cpp:   int latency = GLOBAL_DEF("audio/output_latency", 25);
drivers/alsa/audio_driver_alsa.cpp:     int latency = GLOBAL_DEF("audio/output_latency", 25);
drivers/xaudio2/audio_driver_xaudio2.cpp:       int latency = GLOBAL_DEF("audio/output_latency", 25);
drivers/pulseaudio/audio_driver_pulseaudio.cpp: int latency = GLOBAL_DEF("audio/output_latency", 25);
drivers/rtaudio/audio_driver_rtaudio.cpp:       int latency = GLOBAL_DEF("audio/output_latency", 25);
platform/osx/audio_driver_osx.cpp:      int latency = GLOBAL_DEF("audio/output_latency", 25);
platform/haiku/audio_driver_media_kit.cpp:      int latency = GLOBAL_DEF("audio/output_latency", 25);
```

I think it's a safe bet to let the `25` live.

Comment (akien-mga) [inline platform/android/audio_driver_jandroid.cpp]: It looks like this code has been like this forever (since the first commit). I'd say let's just drop the hardcoded `latency = 50` replacement and see how it goes.

Comment (Rubonnek) [inline platform/android/audio_driver_jandroid.cpp]: Will do.

File: modules/theora/video_stream_theora.cpp (hunk 4 above)
Comment (Rubonnek) [inline modules/theora/video_stream_theora.cpp]: Should I add a comment here too?

Comment (akien-mga) [inline modules/theora/video_stream_theora.cpp]: No I guess it's fine, it was likely a bug fix (if not, git blame will tell the next interested guy the little we know :p).

File: servers/visual/shader_language.cpp [hunk 8]
```diff
@@ -1931,7 +1931,8 @@ bool ShaderLanguage::_validate_function_call(BlockNode *p_block, OperatorNode *p
 		}
 
 		if (!fail) {
-			p_func->return_cache = pfunc->return_type;
+			if (r_ret_type)
```
Comment (akien-mga) [inline servers/visual/shader_language.cpp]: That shouldn't be here, it seems you merged your own commit with 834112772b9cb1f7c1ca04b43ab26e23acb149bc (which explains why @reduz is shown as commit author instead of you).

Comment (Rubonnek) [inline servers/visual/shader_language.cpp]: Whoa, I didn't know I could amend someone else's commit XD

Fixed.

File: modules/theora/video_stream_theora.cpp (hunk 4 above)
Comment (Rubonnek) [inline modules/theora/video_stream_theora.cpp]: Haha :P
